vyper/semantics/types/function.py

```diff
@@ -24,7 +24,53 @@
 from vyper.semantics.types.shortcuts import UINT256_T
 from vyper.semantics.types.subscriptable import TupleT
 from vyper.semantics.types.utils import type_from_abi, type_from_annotation
-from vyper.utils import OrderedSet, keccak256
+from vyper.utils import MemoryPositions, OrderedSet, keccak256, mkalphanum
+
+FunctionSignatures = Dict[str, "ContractFunctionT"]
+
+
+@dataclass
+class FunctionArg:
+    name: str
+    typ: VyperType
+    ast_source: Optional[vy_ast.VyperNode] = None
+    default_value: Optional[vy_ast.VyperNode] = None
+
+
+@dataclass
+class FrameInfo:
```

Comment:
Do we want to do away with `FrameInfo` entirely and move its attributes into `FunctionIRInfo`?

Reply:
Oh hmm, is frame_info set at the same time as the other IR attributes? If so then yes

Reply:
no, it is set at codegen whereas the other IR attributes are set at semantic analysis stage.